File: 14-longest-common-prefix/longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        
        string result = strs[0];

        for(int i=1;i<strs.size();i++){

            int j=0;
            while(strs[i][j] == result[j] && j<strs[i].length() && j<result.length()){
                j++;
            }
            if(j==0) return "";
            result=result.substr(0,j);
        }

        return result;
    }
};

File: 14-longest-common-prefix/longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        
        StringBuilder sb = new StringBuilder(strs[0]);

        int n=strs.length;

        for(int i=1;i<n;i++){
            int j=0;
            
            while( j<sb.length() && j<strs[i].length() && sb.charAt(j)==strs[i].charAt(j) ){
                j++;
            }
            if(j==0) return "";

            sb=new StringBuilder(sb.substring(0,j));
        }
        return sb.toString();
    }
}